### PR TITLE
add thog enterprise detector for web keys

### PR DESCRIPTION
**Pull request by zubairk14**
Look for thog-key, thog-secret, and `word-word-word.word.word.trufflehog.org`. thog scanner keys supported in a separate PR

### CODE REVIEW

File: proto/detectors.proto [hunk 1]
```diff
@@ -929,6 +929,7 @@ enum DetectorType {
   Prefect = 918;
   Docusign = 919;
   Couchbase = 920;
+  TrufflehogEnterprise = 921;
```
Comment (dustin-decker) [inline proto/detectors.proto:932]: run `make protos`

File: pkg/detectors/trufflehogenterprise/trufflehogenterprise.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	hostnameMatches := hostnamePat.FindAllStringSubmatch(dataStr, -1)
+
+	for _, keyMatch := range keyMatches {
+		if len(keyMatch) != 1 {
+			continue
+		}
+		resKeyMatch := strings.TrimSpace(keyMatch[0])
+
+		for _, secretMatch := range secretMatches {
+
```
Comment (ahrav) [inline pkg/detectors/trufflehogenterprise/trufflehogenterprise.go:50]: nit/question: Is this blank line here on purpose?

File: pkg/detectors/trufflehogenterprise/trufflehogenterprise.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+						continue
+					}
+
+					req.Header.Add("Accept", "application/vnd.trufflehogenterprise+json; version=3")
+					req.Header.Add("X-Thog-Secret", resSecretMatch)
+					req.Header.Add("X-Thog-Key", resKeyMatch)
+
+					res, err := client.Do(req)
+					if err == nil {
+
```
Comment (ahrav) [inline pkg/detectors/trufflehogenterprise/trufflehogenterprise.go:81]: ditto re: blank line

**Review by ahrav — APPROVED** [review]
lgtm, nice job

File: pkg/detectors/trufflehogenterprise/trufflehogenterprise.go (hunk 2 above)
Comment (zubairk14) [reply pkg/detectors/trufflehogenterprise/trufflehogenterprise.go:50]: accident I think

**Comment by bugbaba** [comment]
JFYI, The detector wasn't added in [`pkg/engine/defaults.go`](https://github.com/trufflesecurity/trufflehog/blob/main/pkg/engine/defaults.go)

**Pull request merged**